gprofiler/system_metrics.py

```diff
@@ -0,0 +1,91 @@
+import statistics
+import time
+from collections import deque
+from threading import Event, Thread
+from typing import Deque, List, Optional, Tuple
+
+import psutil
+
+from gprofiler.exceptions import ThreadStopTimeoutError
+
+DEFAULT_POLLING_INTERVAL_SECONDS = 5
+STOP_TIMEOUT_SECONDS = 30
+
+
+class SystemMetricsMonitor:
+    def __init__(self, max_memory_poll_age_seconds: int, polling_rate_seconds: int = DEFAULT_POLLING_INTERVAL_SECONDS):
+        self._polling_rate_seconds = polling_rate_seconds
+        self._cpu_count = psutil.cpu_count() or 1
```

Comment:
`or 1`?

Reply:
Taken from `psutil. cpu_percent`, I suppose it can fail and return 0 or `None`

Reply:
I prefer 0 in this case (or None. Anything else we can mark as an error value)